packages/dds/spanner/src/spannerDeltaHandler.ts

```diff
@@ -0,0 +1,54 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import { IDeltaHandler } from "@fluidframework/datastore-definitions";
+
+/**
+ * Handles incoming and outgoing deltas for the Spanner distributed data structure.
+ *
+ * Needs to be able to process v1 and v2 ops
+ */
+export class SpannerDeltaHandler implements IDeltaHandler {
+	private newHandler: IDeltaHandler | undefined;
+	public constructor(
+		private readonly oldHandler: IDeltaHandler,
+		public readonly migrateFunction: (
```

Comment:
Why does this live down here rather than in the `Spanner` directly?  Same question for why the `migrate` method lives on the `SpannerDeltaConnection` rather than in the `Spanner` directly.

Reply:
Oh, that code was a complete hack. That's the reason. I haven't thought it through, so this is just an effect of the code being prototype level.